lib/providers/router_provider.dart

```diff
@@ -0,0 +1,119 @@
+import 'package:dears/models/member_role.dart';
+import 'package:dears/pages/chat_list_page.dart';
+import 'package:dears/pages/chat_page.dart';
+import 'package:dears/pages/chat_redirect_page.dart';
+import 'package:dears/pages/details_page.dart';
+import 'package:dears/pages/favorite_page.dart';
+import 'package:dears/pages/home_page.dart';
+import 'package:dears/pages/inquiry_page.dart';
+import 'package:dears/pages/loading_page.dart';
+import 'package:dears/pages/personal_page.dart';
+import 'package:dears/pages/role_selection_page.dart';
+import 'package:dears/pages/search_page.dart';
+import 'package:dears/pages/search_result_page.dart';
+import 'package:dears/providers/planner_page.dart';
+import 'package:dears/providers/user_info_provider.dart';
+import 'package:go_router/go_router.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'router_provider.g.dart';
+
+@riverpod
+GoRouter goRouter(GoRouterRef ref) {
+  final userInfo = ref.watch(userInfoProvider);
+  return GoRouter(
+    redirect: (context, state) {
+      return userInfo.maybeWhen(
+        data: (user) {
+          if (user.uuid == null) {
+            if (state.matchedLocation != "/select-role") {
+              return "/select-role";
+            }
+          } else {
+            if (state.matchedLocation == "/") {
+              if (user.role == MemberRole.customer) {
+                return "/";
+              } else if (user.role == MemberRole.weddingPlanner) {
+                return "/planner";
+              }
```

Comment:
dart의 enum은 exhaustiveness를 지원합니다. 다음과 같이 `switch` expression으로 변경할 수 있습니다.

```suggestion
              return switch (user.role) {
                MemberRole.customer => "/",
                MemberRole.weddingPlanner => "/planner",
              };
```